src/poll_manager.ts

```diff
@@ -0,0 +1,149 @@
+import type { StreamChat } from './client';
+import type {
+  CreatePollData,
+  DefaultGenerics,
+  ExtendableGenerics,
+  PollResponse,
+  PollSort,
+  QueryPollsFilters,
+  QueryPollsOptions,
+} from './types';
+import { Poll } from './poll';
+import { FormatMessageResponse } from './types';
+import { formatMessage } from './utils';
+
+export class PollManager<SCG extends ExtendableGenerics = DefaultGenerics> {
+  private client: StreamChat<SCG>;
+  private pollCache = new Map<string, Poll<SCG>>();
+  private unsubscribeFunctions: Set<() => void> = new Set();
+
+  constructor({ client }: { client: StreamChat<SCG> }) {
+    this.client = client;
+  }
+
+  get data(): Map<string, Poll<SCG>> {
+    return this.pollCache;
+  }
+
+  public fromState = (id: string) => {
```

Comment:
```suggestion
  public fromCache = (id: string) => {
```

Reply:
Deliberated quite a bit over this as well, but in the end I decided to go with `fromState` since the `cache` is the manager's state in some sense. It feels kind of wrong exposing something called `fromCache` to integrators since it might lead them to believe it's something else, but I might just be overthinking this :D

Reply:
I'm worried that this might be a bit misleading either way as integrators might think that manager has a state which they can subscribe to as the API becomes the norm across the client though I understand your point too.

Reply:
Maybe something like `byId` would also be good ? Not implying anything specifically that way, we're just saying that we're asking for a `poll` with a certain ID. What do you think ?